src/Orchard/Mvc/ModelBinders/BooleanBinderProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Mvc;
+
+namespace Orchard.Mvc.ModelBinders {
+    public class BooleanBinderProvider : IModelBinderProvider, IModelBinder {
+
+        public IEnumerable<ModelBinderDescriptor> GetModelBinders() {
+            return new[] {
+                             new ModelBinderDescriptor {
+                                                           ModelBinder = this,
+                                                           Type = typeof(bool)
+                                                       }
+                         };
+        }
+
+        public object BindModel(ControllerContext controllerContext, ModelBindingContext bindingContext) {
+            var value = false;
+            var requestBooleanValue = controllerContext.HttpContext.Request[bindingContext.ModelName].Split(',')[0]; //Html.CheckBox and Html.CheckBoxFor return "true,false" string
+            if (!bool.TryParse(requestBooleanValue.Split(',')[0], out value)) {
```

Comment:
I think `requestBooleanValue.Split(',')[0]` is unnecessary here as it was already split in the previous statement

Reply:
Yes, you are right. I Removed it; it is mergeable now. Thanks.